### ISSUE TITLE
KIC: api host-bind port (APIServerPort=0)

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#6398]
`💣  Failed to get driver URL: api host port binding: getting host-bind port '\x00' for container ID "minikube": exit status 1`

Some some reason the docker container has a port 0:

```json
            "ExposedPorts": {
                "0/tcp": {},
                "22/tcp": {},
                "8443/tcp": {}
            },
```
```json
            "Ports": {
                "0/tcp": null,
                "22/tcp": [
                    {
                        "HostIp": "127.0.0.1",
                        "HostPort": "32771"
                    }
                ],
                "8443/tcp": [
                    {
                        "HostIp": "127.0.0.1",
                        "HostPort": "32770"
                    }
                ]
            },
```

```
Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:48:43 2019
  OS/Arch:          linux/amd64
  Experimental:     true

```

**Comment by afbjorklund** [comment]
It seemed to be missing logging (?), but here is the command after adding some:

`I0125 10:12:55.621565   23568 oci.go:339] Running docker [run --cpus=2 --memory=2000mb -d -t --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume /var -v /lib/modules:/lib/modules:ro --hostname minikube --name minikube --label io.x-k8s.kic.cluster=minikube --label io.k8s.sigs.kic.role= --expose 0 --publish=127.0.0.1::8443 --publish=127.0.0.1::22 gcr.io/k8s-minikube/kicbase:v0.0.2@sha256:8f531b90901721a7bd4e67ceffbbc7ee6c4292b0e6d1a9d6eb59f117d57bc4e9]`

So something is broken with d.NodeConfig.APIServerPort, even though CPU/Memory works

**Comment by afbjorklund** [comment]
Looks like #6397 fixes this issue

**Comment by priyawadhwa** [comment]
@afbjorklund can this issue be closed now that #6397 has merged?

**Comment by afbjorklund** [comment]
@priyawadhwa : yes, seems to work now.

Unfortunately that PR totally lacked context...

"Fixes a few things"
